crates/router_derive/src/macros/api_error/helpers.rs

```diff
@@ -227,11 +252,25 @@ pub(super) fn check_missing_attributes(
 }
 
 /// Get all the fields not used in the error message.
-pub(super) fn get_unused_fields(fields: &Fields, message: &str) -> Vec<Field> {
+pub(super) fn get_unused_fields(
+    fields: &Fields,
+    message: &str,
+    ignore: &std::collections::HashSet<String>,
+) -> Vec<Field> {
     let fields = match fields {
         syn::Fields::Unit => Vec::new(),
         syn::Fields::Unnamed(_) => Vec::new(),
-        syn::Fields::Named(fields) => fields.named.iter().cloned().collect(),
+        syn::Fields::Named(fields) => fields
+            .named
+            .iter()
+            .filter(|field| {
+                field
+                    .ident
+                    .as_ref()
+                    .map_or(true, |idn| ignore.contains(&idn.to_string()))
```

Comment:
If it makes it any simpler, you can also update the filtering condition (after the match block) to be:

```rust
!(message.contains(&field_name) && ignore.contains(&field_name))
```